zkvm/src/types.rs

```diff
@@ -136,6 +139,43 @@ impl ScalarWitness {
     }
 }
 
+impl Neg for ScalarWitness {
+    type Output = ScalarWitness;
+
+    fn neg(self) -> ScalarWitness {
+        match self {
+            ScalarWitness::Integer(a) => ScalarWitness::Integer(-a),
+            ScalarWitness::Scalar(a) => ScalarWitness::Scalar(-a),
+        }
+    }
+}
+impl Add for ScalarWitness {
+    type Output = ScalarWitness;
+
+    fn add(self, rhs: ScalarWitness) -> ScalarWitness {
+        match (self, rhs) {
+            (ScalarWitness::Integer(a), ScalarWitness::Integer(b)) => {
+                let res = a + b;
```

Comment:
This abuses `SignedInteger`'s not implementing overflow checks internally inside `Add` implementation. We'll need to fix it, but we can do that separately from this PR and get back to this line later.

Reply:
Let's merge this checked arithmetic for `SignedInteger` first and then simplify this code as these checks will be done inside SignedInteger Add/Mul implementations. https://github.com/interstellar/slingshot/pull/106